src/V3Randomize.cpp

```diff
@@ -1272,6 +1272,25 @@ class RandomizeVisitor final : public VNVisitor {
                 }
             }
             return stmtsp;
+        } else if (const auto* const unionDtp = VN_CAST(memberp ? memberp->subDTypep()->skipRefp()
+                                                                : exprp->dtypep()->skipRefp(),
+                                                        UnionDType)) {
+            AstNodeStmt* stmtsp = nullptr;
+            if (!unionDtp->packed()) {
+                unionDtp->v3warn(E_UNSUPPORTED, "Unsupported: IEEE 1800-2023: Unpacked unions "
+                                                "shall not be declared as rand or randc.");
+                return nullptr;
+            }
+            AstMemberDType* firstMemberp = unionDtp->membersp();
+            AstNodeStmt* randp = nullptr;
+            randp = newRandStmtsp(fl, stmtsp ? exprp->cloneTree(false) : exprp, nullptr, offset,
+                                  firstMemberp);
+            if (stmtsp) {
+                stmtsp->addNext(randp);
+            } else {
+                stmtsp = randp;
+            }
```

Comment:
`stmtsp` is always null here before `stmtsp = randp;`.

Reply:
thanks :)